File: js/emit_args.js
function
emit_args(u, p)
{
	var v = {type:PAREN, a:[], small_font:u.small_font};

	p = cdr(p);

	if (iscons(p)) {
		emit_expr(v, car(p));
		p = cdr(p);
		while (iscons(p)) {
			emit_roman_text(v, ",");
			emit_expr(v, car(p));
			p = cdr(p);
		}
	}

	emit_update_subexpr(v);

	u.a.push(v);
}

File: app/emit_args.c
#include "app.h"

void
emit_args(struct atom *p)
{
	int t;

	emit_thin_space();

	p = cdr(p);

	if (!iscons(p)) {
		emit_roman_char('(');
		emit_roman_char(')');
		return;
	}

	t = tos;

	emit_expr(car(p));

	p = cdr(p);

	while (iscons(p)) {
		emit_roman_char(',');
		emit_expr(car(p));
		p = cdr(p);
	}

	emit_update_list(t);

	emit_update_subexpr();
}
